webserver/views/datasets.py

```diff
@@ -335,6 +332,22 @@ def recording_info(mbid):
         return jsonify(error=str(e)), 404
 
 
+@datasets_bp.route("/metadata/recording/<uuid:mbid>")
+@login_required
+def recording_info(mbid):
+    """Endpoint for getting information about recordings (title and artist)."""
+    return _get_recording_info_for_mbid(mbid)
+
+
+@datasets_bp.route("/metadata/dataset/<uuid:dataset_id>/<uuid:mbid>")
+def recording_info_in_dataset(dataset_id, mbid):
+    """Endpoint for getting information about recordings (title and artist), for the
+     case when user is not logged in"""
+    if not db.dataset.check_recording_in_dataset(dataset_id, mbid):
+        return jsonify(error="Recording not found in the dataset"), 404
```

Comment:
Would `raise NotFound("Recording not found in the dataset")` be better here?

Reply:
I did originally suggest this, but there's a reason I didn't make the change (which I can't completely remember)...

We can use our `API*` exception classes to return json errors. We still have an issue with these endpoints that a call to them with bad uuids (for example) will return a flask 404 which will be html - same issue as https://github.com/metabrainz/acousticbrainz-server/pull/232 except we don't have a url namespace to filter on.
Two options would be:
 * put api endpoints which are used only by the interface under `/api` somwhere
 * see if it's possible to make a decorator that you can say "return any exception from here in html"

Reply:
I think making these new views API endpoints (first option) would be a better idea.

Reply:
Maybe this should be done in a new patch, this has been standing for too long anyways.

Reply:
[AB-319](https://tickets.metabrainz.org/browse/AB-319)